pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -985,13 +1046,21 @@ def update_device_type_if_training_type_plugin_passed(self) -> None:
                     self._device_type = DeviceType.TPU
                 elif self.use_gpu:
                     self._device_type = DeviceType.GPU
+                elif self.use_hpu:
+                    self._device_type = DeviceType.HPU
             else:
                 if self.has_ipu:
                     self._device_type = DeviceType.IPU
                 elif self.has_tpu:
                     self._device_type = DeviceType.TPU
                 elif self.has_gpu:
                     self._device_type = DeviceType.GPU
+                elif self.has_hpu:
+                    self._device_type = DeviceType.HPU
+
+    def update_device_type_if_hpu_plugin(self) -> None:
```

Comment:
Is this function used ?

Reply:
Will check and remove if not required. But there was an issue with devicetype